agr_mvs/turno.py

```diff
@@ -0,0 +1,149 @@
+from datetime import datetime
+from agr_mvs.ticket import Ticket
+import math
+
+# Constante que nos expresa la longitud del turno que es 5'5 horas.
+HORAS_TURNO = 5.5
+
+class Turno:
+    Id: int                      # Identificador del Turno
+    Horas_Turno: list            # Mediante este array introducimos dos datetime que nos indican el inicio y el fin del turno
+    Tickets: list                # Tickets dentro de un Turno.
+    Fecha:  datetime             # Fecha del Turno
+    µ: float                     # Clientes que son atendidos por unidad de tiempo (hora)
+    λ: float                     # Clientes que llegan por unidad de tiempo (hora)
+    S: float                     # Numero de TPV que tenemos disponibles (Nos los aporta la tienda como dato)           
+            
+
+    
+    def __init__(self, id, horas_turno, S, Fecha, µ=None, λ=None, p=None):
+        self.Id = id
+        self.Horas_Turno = horas_turno
+        self.Fecha = Fecha
+        self.Tickets = []
+        self.µ = µ
+        self.S = S
+        self.λ = λ
+
+    def __str__(self):
+        cadena = "-----------Objeto tipo Turno-----------\n"
+        print(cadena)
+        print("id =", self.Id)
+        print("Tipo_turno =", self.Horas_Turno)
+        print("Fecha =", self.Fecha)
+        print("µ =", self.µ)
+        print("λ =", self.λ)
+        print("S =", self.S)
+        print("Tickets =", self.Tickets)
+        print("\n")
+        return cadena
+    # Añadimos un Ticket a la lista de Tickets del Turno
+    def addTicket(self, ticket):
+        if (self.Fecha.date() == ticket.FechaTicket.date()):
+                if (ticket.FechaTicket.timestamp() >= self.Horas_Turno[0].timestamp() and ticket.FechaTicket.timestamp() <= self.Horas_Turno[1].timestamp()):
+                    self.Tickets.append(ticket)
+        else:
+            # Excepción del tipo IndexError en caso de que no se pueda añadir el Ticket
+            raise IndexError("El Ticket no pertenece al Turno")
+            
+    # Método get para obtener el nombre de una tienda
+    def getNombre(self):
+        return self.Nombre
+    
+    def getµ(self):
+        return self.µ
+
+    def getλ(self):
+        return self.λ
+
+    def getS(self):
```

Comment:
Ya lo he puesto varias veces, pero lo vuelvo a poner. Esto simplemente no lo usas. Incluid solo en el API las funciones que vayáis a usar.

Reply:
Eliminado todos los get y set